### PR TITLE
Add log1p

### PR DESCRIPTION
**Pull request by tillahoffmann**
This PR adds the `log1p` function and partially addresses #3682. It also uses `log1p` in the computation of the sigmoid cross entropy to improve numerical stability (I had issues with rare events).

### CODE REVIEW

File: tensorflow/core/kernels/cwise_op_log1p.cc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
```
Comment (vrv) [inline tensorflow/core/kernels/cwise_op_log1p.cc]: Probably 2016 now

**Review by vrv — COMMENTED** [review]
Very nice!!

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Comment by vrv** [comment]
Yes, seems reasonable to me.  I don't know why the tests are written like that anyway.

**Comment by tillahoffmann** [comment]
Could just change them to perform an equivalent computation using numpy routines.

**Comment by vrv** [comment]
@zakariahaque can you comment?

**Comment by zakariahaque** [comment]
@tillahoffmann, thanks for the change.  Feel free to update the loss in the tests.
@vrv all the test needs to check is that the correct loss was used for that type of head/target_column.

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Review by vrv — APPROVED** [review]
